### DISCUSSION TITLE
V3: SVG-Sprite images disappear on navigating

### DISCUSSION THREAD

**Discussion by renky** [livewire/livewire#5843] [7. Bugs (archived)]
we use icons generated via svg sprite that are included in the page like this:

```
<svg class="w-4 h-4">
    <use xlink:href="file-open"/>
</svg>
```

after navigation all icons disappear, also adding wire:ignore to the svg doesn't help in that case.
![image](https://github.com/livewire/livewire/assets/6528960/603be273-df82-4121-a8f2-3b2eb8af7c2c)
changes to
![image](https://github.com/livewire/livewire/assets/6528960/290b85f6-62ab-4569-91ba-1d0d9c58f129)

don't have an idea how to analyse this more

the sprite is generated via vite-plugin:
```
import svgSpritePlugin from 'vite-plugin-svg-sprite-component'

...

plugins: [
    ....
    svgSpritePlugin({
      symbolId: (name) => name
    })
  ]
```

the svg-icons are then placed in /resources/svgs
for example /resources/svgs/file-open.svg

and finally they are imported in JS app.js:
```
import.meta.globEager('../svgs/*.svg')
```

**Comment by joshhanley** [comment]
@renky thanks for reporting! Any chance you could open a PR with a failing test? Thanks!

**Reply by renky** [reply]
Unfortunatelly I don't have a clue how we could force this behaviour by an automated test... The HTML itself looks correct, even after the bug appeared, and I think it is a rendering problem in Browsers after morphing ht code - even if it doesn't change. 

Before first presentation about V3, i made some navigation-tests with morphDom (wanted to create a soft-load function on my own). I cancelled this try, because it didn't work in several cases and I also remember that one of it was the svgs disappearing...

I also think this isn't a v3/livewire problem - but maybe we can find a solution for livewire to handle it?! At least for me it would be a reason to not use the navigation function and btw: when just updating livewire-components it seems like this problem doesn't exist - only with navigation it started... I also found related blog/question-entries that seems to be similar with react or even native JS:

https://stackoverflow.com/questions/50771280/react-svg-disappearing-when-component-rerenders
https://stackoverflow.com/questions/35167370/svg-image-is-added-to-dom-but-isnt-visible-unless-touched-in-firebug
...

the big question for me is: why is it working inside a Livewire-component (for example I use this type of icons in a table action column... switching pages of the table works without any problem - maybe this works, because also the html for the icons is replaced and new, or neavigation doesn't work, because it is replacing to much - I don't have a clue...

**Reply by mauricekindermann** [reply]
I just hit this bug in production. It's absoutely bizarre, seems to be new to my project, never seen it before and suddenly a week ago it's started happening seemingly at random (usually on wire navigate but also on refresh, though it always works on hard refresh). I'm hoping it's one of those transient Chrome-specific browser bugs which I've come across a few tiems over the years.

**Reply by renky** [reply]
I never found a solution. Finally I removed the svg-sprites completly from my project and use blade-icons now...

**Reply by mauricekindermann** [reply]
I can't do that, that's what I moved away from, I have cards that load 50 at a time, icons can end up in the thousands, page gets too slow, dom gets too heavy. Whereas a single sprite SVGs gets loaded into cache and is magnificent.

As I suspected, I dug deeper, found these: 

https://issues.chromium.org/issues/40140856
https://issues.chromium.org/issues/41323246

Similar bug fixed in webkit in 2012 (<use> recalc issue):
https://bugs.webkit.org/show_bug.cgi?id=74392

GitLab even have a writeup on a similar problem and workaround:
https://gitlab.com/gitlab-org/gitlab/-/merge_requests/38304

**Reply by mauricekindermann** [reply]
I have noticed a few weeks later - this doesn't happen anymore, or at least I am unable to make it fail. 

- Tried disabled cache with CMD + F5 over and over
- Tried regular refresh with cache

I tried all the combos. Before it was failing often, now I cannot make it  fail.

Fingers crossed it's patched/solved though some transient chrome bug.
